src/components/NewComment.jsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from "react";
+import { postComment } from "../utils/axiosFunctions";
+import CommentMessage from "./CommentMessage";
+
+export default function NewComment({
+  article_id,
+  username,
+  setCurrentComments,
+}) {
+  const [newComment, setNewComment] = useState(null);
+  const [tempComment, setTempComment] = useState("");
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const [commentMessage, setCommentMessage] = useState(null);
+  const [lockButton, setLockButton] = useState(false);
+
+  useEffect(() => {
+    if (newComment !== null && formSubmitted) {
+      postComment(article_id, { username: username, body: newComment }).then(
+        (postedComment) => {
+          setCommentMessage("success");
+          setLockButton(false);
+          setCurrentComments((currentComments) => {
+            const updatedComments = [postedComment, ...currentComments];
+            return updatedComments;
+          });
+        }
+      );
+    }
+    setNewComment(null);
+  }, [formSubmitted]);
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    switch (true) {
```

Comment:
I have not seen this syntax before (passing `true` to a switch block) this is wild, but all good 👍